litgpt/prompts.py

```diff
@@ -201,15 +201,43 @@ def apply(self, prompt: str, **kwargs: str) -> str:
 
 
 class Llama3(PromptStyle):
-    def apply(self, prompt: str, **kwargs: str) -> str:
+    def apply(self, prompt: Union[str, List[Dict[str, str]]], **kwargs: str) -> str:
         # https://github.com/meta-llama/llama3/blob/359887376f0aaf30e433f23e25df858d8c2a9833/llama/tokenizer.py#L202-L229
-        return (
-            "<|begin_of_text|><|start_header_id|>system<|end_header_id|>\n\n"
-            "You are a helpful assistant.<|eot_id|>\n"  # The system prompt is optional
-            "<|start_header_id|>user<|end_header_id|>\n\n"
-            f"{prompt}<|eot_id|>\n"
-            "<|start_header_id|>assistant<|end_header_id|>\n\n"
-        )
+        if isinstance(prompt, str):
+            return (
+                "<|begin_of_text|><|start_header_id|>system<|end_header_id|>\n\n"
+                "You are a helpful assistant.<|eot_id|>\n"  # The system prompt is optional
+                "<|start_header_id|>user<|end_header_id|>\n\n"
+                f"{prompt}<|eot_id|>"
+                "<|start_header_id|>assistant<|end_header_id|>\n\n"
+            )
+        elif isinstance(prompt, list):
+            template = (
+                "<|begin_of_text|><|start_header_id|>system<|end_header_id|>\n\n"
+                "{system}<|eot_id|>\n"  # The system prompt is optional
+            )
+            user_template = (
+                "<|start_header_id|>user<|end_header_id|>\n\n"
+                "{user_msg}<|eot_id|>"
```

Comment:
removed `\n` from here since the official Llama3 model card doesn't have it. 
<img width="831" alt="Screenshot 2024-06-13 at 11 09 43 AM" src="https://github.com/Lightning-AI/litgpt/assets/21018714/b3f824fa-f617-4a2b-a550-40ec9befd03f">